### PR TITLE
ci: Run compatibility test ignoring lockfile

### PR DESCRIPTION
**Pull request by legobeat**
This adds an additional run of tests against a state where `yarn.lock` is removed prior to the installation. This is intended to catch e.g. regressions and mislabeled breakage in updated transitive dependencies.

This will map more closely to what users get when installing the released package from registry.

Left out the intermediate versions of node as an optimization. The normal cache is still used since significant overlap is expected.

### Potential future improvements

- Run test against built package under another package manager (yarn v1 and/or npm)
  - Ensure there aren't implicit dependencies on yarn berry
  - Would allow throwing on breakages in `engines` without adding a plugin

### CODE REVIEW

File: .github/workflows/build-lint-test.yml [hunk 1]
```diff
@@ -122,3 +122,28 @@ jobs:
             echo "Working tree dirty at end of job"
             exit 1
           fi
+  compat_test:
```
Comment (Mrtenz) [inline .github/workflows/build-lint-test.yml:125]:
```suggestion

  compatibility-test:
```

File: .github/workflows/build-lint-test.yml [hunk 2]
```diff
@@ -122,3 +122,28 @@ jobs:
...
+    name: Compatibility Test
```
Comment (Mrtenz) [inline .github/workflows/build-lint-test.yml:126]:
```suggestion
    name: Compatibility test
```

**Review by Mrtenz — CHANGES_REQUESTED** [review]
Not sure I fully understand the need for this, but either way here's some nits. 😅

**Review by Mrtenz — APPROVED** [review]
Makes sense!

**Review by mcmire — APPROVED** [review]
Looks good!

**Pull request merged**